edge-runtime/lib/middleware.ts

```diff
@@ -67,6 +67,10 @@ export function mergeMiddlewareCookies(middlewareResponse: Response, lambdaReque
   const middlewareCookies = middlewareResponse.headers.get('x-middleware-set-cookie')
 
   if (middlewareCookies) {
+    // Next expects internal headers to be omitted when cookies are set by the middleware
+    // See: https://github.com/vercel/next.js/blob/005db43079c7b59fd8c2594e8362761dc4cb3211/test/e2e/app-dir/app-middleware/app-middleware.test.ts#L197-L207
+    middlewareResponse.headers.delete('x-middleware-set-cookie')
```

Comment:
Does final response reaching client (browser/curl etc) would have actual `set-cookie` with this removal? from what I can see `mergeMiddlewareCookies` only sets `cookie` header:
https://github.com/opennextjs/opennextjs-netlify/blob/7795157ec4ab1270acb063abb28ac6fc5ad423a2/edge-runtime/lib/response.ts#L245-L248

so I wonder if this change as-is wouldn't result in clients no longer actually receiving `set-cookie` headers?

Reply:
Setting the `set-cookie` header is done in `addMiddlewareHeaders` which happens right after the highlighted code.
https://github.com/opennextjs/opennextjs-netlify/blob/7795157ec4ab1270acb063abb28ac6fc5ad423a2/edge-runtime/lib/middleware.ts#L53-L59

Side note - Since `x-middleware-set-cookie` is in `INTERNAL_HEADERS` it gets filtered out from the request as soon as it reaches the lambda, so this change is pretty much just for response header cleanliness and the e2e test.